### DISCUSSION TITLE
Weird 404 Not Found exception when rehydrating models

### DISCUSSION THREAD

**Discussion by paulbarby** [livewire/livewire#5099] [1. Help]
I have a model with one primary index and multiple other index's. I was getting a 404 error when livewire tried to rehydrate the model on the next request.

The problem surfaced when I added a list of fields to the eloquent builder query ->first(['id','title','properties','key']);
id title and key are all indexed and id is primary , title and key are varchar(191) while id is a bigint.

I kept getting a 404 exception and Laravel would show me a 404 page in the livewire error modal.

I tracked it down to the rehydration was using the value from the "key" field when it tried to pull the record back in for rehydration. But it was failing as it seemed to be using the the value in the "id" field for the firstOrFail.

I solved this by removing the "key" field form the field list and the problem when away. 
 ->first(['id','title','properties']);

Don't bother replying to this I just though it might help some one else and save them some time.

```
0 \vendor\laravel\framework\src\Illuminate\Queue\SerializesAndRestoresModelIdentifiers.php(102): Illuminate\Database\Eloquent\Builder->firstOrFail()
#1 \vendor\laravel\framework\src\Illuminate\Queue\SerializesAndRestoresModelIdentifiers.php(57): Livewire\HydrationMiddleware\HydratePublicProperties->restoreModel(Object(Illuminate\Contracts\Database\ModelIdentifier))
#2 \vendor\livewire\livewire\src\HydrationMiddleware\HydratePublicProperties.php(145): Livewire\HydrationMiddleware\HydratePublicProperties->getRestoredPropertyValue(Object(Illuminate\Contracts\Database\ModelIdentifier)) 
#3 \vendor\livewire\livewire\src\HydrationMiddleware\HydratePublicProperties.php(58): Livewire\HydrationMiddleware\HydratePublicProperties::hydrateModel(Array, 'instance', Object(Livewire\Request), Object(App\Http\Livewire\ViewObjectInstance))
#4 \vendor\livewire\livewire\src\LifecycleManager.php(89): Livewire\HydrationMiddleware\HydratePublicProperties::hydrate(Object(App\Http\Livewire\ViewObjectInstance), Object(Livewire\Request))

```

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147
